src/tests/conduit/t_conduit_node_compare.cpp

```diff
@@ -191,31 +194,39 @@ TEST(conduit_node_compare, compare_leaf_numeric)
 TEST(conduit_node_compare, compare_leaf_string)
 {
     const std::string compare_strs[4] = {"I", "me", "You", "tHeM"};
+    char compare_buffs[5][10];
 
     for(index_t fi = 0; fi < 2; fi++)
     {
         NodeDiffFun diff_nodes = NODE_DIFF_FUNS[fi];
 
         for(index_t ci = 0; ci < 4; ci++)
         {
-            std::string comp_str = compare_strs[ci];
-            std::string diff_str = comp_str;
-            {
-                size_t test_len = comp_str.length();
-                diff_str[test_len-1] += 1;
-            }
+            index_t leaf_tid = DataType::CHAR8_STR_ID;
```

Comment:
To help with future debugging -- can you add a comment describing how the offset is being used. Maybe even show the pattern of the data you are comparing?

"heM" == "heM" (or whatever it may be )

Reply:
I've added this with my latest commit.